api/machinemanager/machinemanager.go

```diff
@@ -158,14 +158,17 @@ func (client *Client) UpgradeSeriesPrepare(machineName, series string, force boo
 		Force:  force,
 	}
 	result := params.ErrorResult{}
-	err := client.facade.FacadeCall("UpgradeSeriesPrepare", args, &result)
-	if err != nil {
+	if err := client.facade.FacadeCall("UpgradeSeriesPrepare", args, &result); err != nil {
 		return errors.Trace(err)
 	}
-	if result.Error != nil {
-		return result.Error
-	}
 
+	err := result.Error
+	if err != nil {
+		if params.IsCodeAlreadyExists(err) {
```

Comment:
Why not use `serverError(err)` found in `apiserver/common/errors.go`? Seems to be kind of in a bad spot though?

Reply:
We are using `ServerError` in the API server side. That is what populates the correct code for the "already exists" that is raised in the state package when a lock is already found. We re-hydrate the correct error client-side.

This looks like the correct pattern, though it's the first time I've used it.